chart/values.yaml

```diff
@@ -47,13 +47,20 @@ processor:
     requests:
       cpu: "0.5"
       memory: "500Mi"
-  # See KAPELConfig.py for full details on required configuration.
-  config: |
+  # Config values for the processor initContainer. See KAPELConfig.py for full details on required configuration.
+  config:
+    # Name of the site being reported on
     SITE_NAME: "EXAMPLE-T2"
+    # Uniquely identifying name for the cluster
     SUBMIT_HOST: "k8s.example.org:6443/namespace"
-    #BENCHMARK_VALUE: "15.0"
-    #NAMESPACE: "harvester"
-    #VO_NAME: "atlas"
+    # Namespace in the cluster where pilot jobs run
+    NAMESPACE: "example-namespace"
```

Comment:
I had these required variables commented before so that if you just start up the chart with default config as a basic test, it would explicitly fail, to avoid the risk of accidentally publishing invalid/fake/test records to a production accounting service. Basically forcing the user to explicitly set the required configuration.
The errors should show up clearly during the config initialization, e.g. an exception from the environs package saying that a required setting is missing.  https://github.com/rptaylor/kapel/blob/master/python/KAPELConfig.py#L19

With this change it will run until it gets to the Prom query which would fail for a (hopefully) non-existent namespace, but that might be less clear and less user friendly.

@mwestphall what do you think would be the best way to address this? 
I know Helm can print some post-install output, maybe that could help as a reminder or confirmation to the user.

Reply:
> I know Helm can print some post-install output, maybe that could help as a reminder or confirmation to the user.

Oh I didn't know this was a thing! I think the quick fix of commenting out this value is fine for expediency

Reply:
@rptaylor I think raising a warning or error at the time of install is probably the way to go. If we just want a warning, it looks like [NOTES.txt](https://helm.sh/docs/chart_template_guide/notes_files/) is printed at the end of helm install and can utilize templating, so we could do something like 
```
{{ if not .Values.processor.config.NAMESPACE  -}}
WARNING: processor.config.NAMESPACE is empty in values.yaml. Set it to the namespace where your workload jobs run.
{{- end -}}
```
Which results in a successful chart installation with that warning printed at the end.

Otherwise, we could make the chart install fail using the [`required` template](https://helm.sh/docs/howto/charts_tips_and_tricks/#know-your-template-functions), eg:
```yaml
kind: ConfigMap
data:
  NAMESPACE: {{ required ".Values.processor.config.NAMESPACE must be set" .Values.processor.config.NAMESPACE }}
```
This gets back to the issue of tightly coupling the helm chart validation to the processor container's environment, but it might make sense to do for the "must configure" values that aren't likely to change from chart version to chart version.

Reply:
There are only 4 required values (for which there is no valid safe default IMO if you want to be sure not to pollute the accounting DB with fake default records) - NAMESPACE, SITE_NAME, SUBMIT_HOST, and VO_NAME.

So it could be a nice friendliness feature to print a warning/reminder in NOTES.txt, I don't think tightly coupling is much of a worry just for those 4 required values. The requirement to set them can also be highlighted in the docs if not already. 

Is it okay with you to keep them commented in the default values.yaml then?

Reply:
@rptaylor yeah, I think this approach makes sense. I'll update the PR to re-comment those values and add the NOTES.txt messages first thing on Monday.